mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -664,56 +704,72 @@ abstract class BaseMvRxViewModel<S : MvRxState>(
         lifecycleOwner: LifecycleOwner? = null,
         deliveryMode: DeliveryMode,
         subscriber: (T) -> Unit
     ): Disposable {
-        return observeOn(AndroidSchedulers.mainThread())
-            .resolveSubscription(lifecycleOwner, deliveryMode, subscriber)
+        return resolveSubscription(lifecycleOwner, deliveryMode, subscriber)
+            .toDisposable()
             .disposeOnClear()
     }
 
-    private fun <T : Any> Observable<T>.resolveSubscription(
+    private fun <T : Any> Flow<T>.resolveSubscription(
         lifecycleOwner: LifecycleOwner? = null,
         deliveryMode: DeliveryMode,
         subscriber: (T) -> Unit
-    ): Disposable = if (lifecycleOwner == null || FORCE_DISABLE_LIFECYCLE_AWARE_OBSERVER) {
-        this.subscribe(subscriber)
-    } else {
-        this.subscribeWith(
-            MvRxLifecycleAwareObserver(
-                lifecycleOwner,
-                deliveryMode = deliveryMode,
-                lastDeliveredValue = if (deliveryMode is UniqueOnly) {
-                    if (activeSubscriptions.contains(deliveryMode.subscriptionId)) {
-                        throw IllegalStateException(
-                            "Subscribing with a duplicate subscription id: ${deliveryMode.subscriptionId}. " +
-                                "If you have multiple uniqueOnly subscriptions in a MvRx view that listen to the same properties " +
-                                "you must use a custom subscription id. If you are using a custom MvRxView, make sure you are using the proper" +
-                                "lifecycle owner. See BaseMvRxFragment for an example."
-                        )
-                    }
-                    activeSubscriptions.add(deliveryMode.subscriptionId)
-                    lastDeliveredStates[deliveryMode.subscriptionId] as? T
-                } else {
-                    null
-                },
-                onNext = Consumer { value ->
-                    if (deliveryMode is UniqueOnly) {
-                        lastDeliveredStates[deliveryMode.subscriptionId] = value
-                    }
-                    subscriber(value)
-                },
-                onDispose = {
-                    if (deliveryMode is UniqueOnly) {
-                        activeSubscriptions.remove(deliveryMode.subscriptionId)
-                    }
-                }
-            )
-        )
+    ): Job {
+        val flow = if (lifecycleOwner == null || FORCE_DISABLE_LIFECYCLE_AWARE_OBSERVER) {
+            this
+        } else if (deliveryMode is UniqueOnly) {
+            val lastDeliveredValue: T? = lastDeliveredValue(deliveryMode)
+            this
+                .assertOneActiveSubscription(lifecycleOwner, deliveryMode)
+                .dropWhile { it == lastDeliveredValue }
+                .flowWhenStarted(lifecycleOwner)
+                .distinctUntilChanged()
+                .onEach { lastDeliveredStates[deliveryMode.subscriptionId] = it }
```

Comment:
It introduces a behavior change comparing with previous version.
here is a test demonstrates an issue (Put it into FragmentSubscriberTest):

```
    @Test
    fun testUniqueOnly() {
        val (controller, fragment) = createFragmentInTestActivity<ViewSubscriberFragment>()
        assertEquals(0, fragment.selectSubscribeUniqueOnlyValue)

        fragment.setFoo(1)

        assertEquals(1, fragment.selectSubscribeUniqueOnlyValue)
        assertEquals(2, fragment.selectSubscribeUniqueOnlyCallCount)

        controller.pause()
        controller.stop()
        fragment.setFoo(2)
        fragment.setFoo(1)
        controller.start()
        controller.resume()

        assertEquals(1, fragment.selectSubscribeUniqueOnlyValue)
        assertEquals(3, fragment.selectSubscribeUniqueOnlyCallCount)
    }
```

it passed in master, but fails in current branch

Reply:
@denis-bezrukov This is an interesting case. @elihart @BenSchwab What do you think the expected value here should be? In words, if a user subscribes with uniqueOnly(), if the value changes and changes back while the Fragment is stopped, should it redeliver? Previously, it would but now it wouldn't. My hunch says that the new behavior is better. What do you think?

Reply:
Ha - this is interesting.. I could go either way..

The old behavior may be useful in some cases, say you selectSubscribeUniqueOnly. And the toast shown is set to `true`, then we go to stop, and then we reset the toast, then something else set's it back to `true`. The expectation may be that the toast is shown again.

Reply:
@BenSchwab I don't know. My gut says that the toast shouldn't be shown because the trigger didn't happen when the toast was shown, it could have happened long before.

If we think this is the last open question in this PR, I think we should land it so subsequent work can begin and consider this case more carefully.

Reply:
@gpeal i could also go either way. not sure there is a correct answer here but:
1) we need to make sure to document this very clearly
2) possibly provide a setting or name change besides `uniqueOnly` so users can control this